node-eclib.js

```diff
@@ -191,17 +191,24 @@ ECLib.prototype = {
         // the 6th fragment.
         var done = 0;
 
-        fragmentIds.sort();
+        fragmentIds.sort(function(a, b) {
+            return (a - b);
+        });
+        var len = fragmentIds.length;
+        var recFrags = new Array(len);
 
         // Recover all missing fragments one by one.
-        fragmentIds.forEach(function reconstructEach(id) {
+        fragmentIds.forEach(function reconstructEach(id, index) {
             this.reconstructFragment(availFragments, id,
                     function recon(err, fragment) {
                         if (err) {
-                            callback(err);
+                            return callback(err);
```

Comment:
You cannot return early from a forEach and using this would call the callback multiple times. You would have to use a vanilla for loop or use some array method like `arr.every`